lib/spack/spack/build_systems/hip.py

```diff
@@ -98,6 +98,8 @@ class HipPackage(PackageBase):
     depends_on('hsa-rocr-dev', when='+hip')
     depends_on('hip', when='+hip')
 
+    depends_on('blt@0.3.7:', type='build', when='+hip')
```

Comment:
Ok, everything else looks good but I think I misunderstood this constraint when I was reviewing earlier. Is the idea here that if a HipPackage is using BLT, then it has to use a recent enough version? If that's what we're trying to encode here, then this needs to be 
```suggestion
    conflicts('^blt@:0.3.6', when='+hip')
```

Reply:
I just pushed a fix. But don't merge this yet, wait until my fix here https://github.com/spack/spack/pull/20216/files and https://github.com/LLNL/blt/pull/435/files go in first, and I do a merge

Reply:
@becker33 : All merged, should be ready now! The blt fix https://github.com/LLNL/blt/pull/435 should go in shortly